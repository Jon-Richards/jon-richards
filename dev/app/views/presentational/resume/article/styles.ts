import { css } from '@emotion/core';
import { Fonts } from 'Config/styles/fonts';
import { Dimensions } from 'Config/styles/dimensions';

export const STYLES = {
  article: css`
    display: block;
    margin: 0;
    padding: 0;
    width: auto;
    height: auto;
  `,
  content: css`
    padding: ${Dimensions.whitespace};
  `,
  title: css`
    font-family: ${Fonts.OPEN_SANS.fontFamily};
    font-size: 1.8rem;
    font-weight: ${Fonts.OPEN_SANS.fontStyles.regular.weights.light};
    margin-bottom: ${Dimensions.whitespace};
  `,
}
