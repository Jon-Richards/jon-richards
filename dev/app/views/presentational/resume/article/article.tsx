/** @jsx jsx */

import * as React from 'react';
import { Panel } from 'Views/presentational/panel';
import { jsx } from '@emotion/core';
import { STYLES } from './styles';

export function Article(props: ArticleProps): JSX.Element {
  return (
    <article css={STYLES.article} data-testid="article">
      <Panel>
        <div css={STYLES.content}>
          <h3 data-testid="article__title" css={STYLES.title}>
            {props.title}
          </h3>
          <div data-testid="article__children">
            {props.children}
          </div>
        </div>
      </Panel>
    </article>
  );
}

export type ArticleProps = React.PropsWithChildren<{
  title: string;
}>;
