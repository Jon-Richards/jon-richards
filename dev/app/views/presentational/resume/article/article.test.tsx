import * as React from 'react';
import { Article } from './article';
import { render } from 'enzyme';

describe('The Article component.', () => {
  it('Should match the snapshot.', () => {
    const wrapper = render(
      <Article title="My Article">
        <p data-testid="test-content">Test Content.</p>
      </Article>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('Should render with the supplied title.', () => {
    const wrapper = render(
      <Article title="My Article">
        <p>Test Content.</p>
      </Article>
    );

    const testHook = '[data-testid="article__title"]'

    expect(wrapper.find(testHook).length).toBe(1);
    expect(wrapper.find(testHook).html()).toBe('My Article');
  });

  it('Renders with the supplied child content.', () => {
    const wrapper = render(
      <Article title="My Article">
        <p>Test Content.</p>
      </Article>
    );

    const testHook = '[data-testid="article__children"]'

    expect(wrapper.find(testHook).length).toBe(1);
    expect(wrapper.find(testHook).html()).toBe('<p>Test Content.</p>');
  });
});
