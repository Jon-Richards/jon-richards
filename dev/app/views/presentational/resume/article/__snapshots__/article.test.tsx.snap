// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Article component. Should match the snapshot. 1`] = `
.emotion-4 {
  display: block;
  margin: 0;
  padding: 0;
  width: auto;
  height: auto;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 100%;
  height: auto;
  position: relative;
  padding: 1px;
  margin: 0rem;
  overflow: hidden;
  background-color: hsla(0,0%,14%,1);
  background-image: linear-gradient( 170deg, hsla(0,0%,30%,1), hsla(0,0%,10%,1), hsla(0,0%,17%,1) );
  box-shadow: 0.4rem 0.8rem 2rem 0rem hsla(0,0%,0%,0.5);
}

.emotion-2 {
  display: block;
  width: 100%;
  height: 100%;
  margin: 0rem;
  padding: 0rem;
  background-color: hsla(0,0%,14%,1);
  background-image: linear-gradient( 170deg, hsla(0,0%,14%,1), hsla(0,0%,8%,1) );
}

.emotion-1 {
  padding: 2rem;
}

.emotion-0 {
  font-family: "Open Sans",sans-serif;
  font-size: 1.8rem;
  font-weight: 300;
  margin-bottom: 2rem;
}

<article
  class="emotion-4"
  data-testid="article"
>
  <div
    class="emotion-3"
  >
    <div
      class="emotion-2"
    >
      <div
        class="emotion-1"
      >
        <h3
          class="emotion-0"
          data-testid="article__title"
        >
          My Article
        </h3>
        <div
          data-testid="article__children"
        >
          <p
            data-testid="test-content"
          >
            Test Content.
          </p>
        </div>
      </div>
    </div>
  </div>
</article>
`;
